src/osd/modules/debugger/debuggdbstub.cpp

```diff
@@ -682,6 +682,9 @@ static std::string escape_packet(const std::string src)
 	result.reserve(src.length());
 	for ( char ch: src )
 	{
+		if ( ch == '\n' ) // don't let socket convert line endings and messing up the checksum
+			continue;
+
```

Comment:
Wouldn’t the proper solution to this be to change the calls to `m_socket.puts(...)` to use `m_socket.write(...)` instead to avoid text conversion altogether?  This looks like working around the issue rather than actually fixing it.

Reply:
yeah, probably. wasn't familiar with the socket interface :)